django/http/response.py

```diff
@@ -141,11 +154,15 @@ def reason_phrase(self, value):
     def charset(self):
         if self._charset is not None:
             return self._charset
-        content_type = self.get("Content-Type", "")
-        matched = _charset_from_content_type_re.search(content_type)
-        if matched:
-            # Extract the charset and strip its double quotes
-            return matched["charset"].replace('"', "")
+        # The Content-Type header may not yet be set, because the charset is
+        # being inserted *into* it.
+        if content_type := self.headers.get("Content-Type", ""):
```

Comment:
Default is unnecessary:
```suggestion
        if content_type := self.headers.get("Content-Type"):
```